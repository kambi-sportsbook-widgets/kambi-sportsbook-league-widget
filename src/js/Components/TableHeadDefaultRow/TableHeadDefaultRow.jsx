import React, { PropTypes } from 'react';
import { translationModule } from 'kambi-widget-core-library';
import styles from './TableHeadDefaultRow.scss';

const t = translationModule.getTranslation.bind(translationModule);

const TableHeadDefaultRow = ({ title, columns, hiddenMode }) => {
   return (
      <tr className={styles.general}>
         <th colSpan="2" className="title">{title}</th>
         {columns.map((column, i) => <th key={i} className="column-name" title={t(column.name)}>{hiddenMode ? '' : column.short}</th>)}
         <th className="margin" />
      </tr>
   );
};

TableHeadDefaultRow.propTypes = {
   /**
    * Widget's title
    */
   title: PropTypes.string,

   /**
    * Statistics and/or outcomes columns to be displayed
    */
   columns: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      short: PropTypes.string
   })).isRequired,

   /**
    * Should mobile header be displayed in widget's hidden mode?
    */
   hiddenMode: PropTypes.bool
};

TableHeadDefaultRow.defaultProps = {
   title: '',
   hiddenMode: false
};

export default TableHeadDefaultRow;
