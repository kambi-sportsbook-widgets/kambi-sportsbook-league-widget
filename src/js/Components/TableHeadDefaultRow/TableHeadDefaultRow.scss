@import '../TableHeadShared';

:local(.general) {
   @extend .table-head-row;

   th {
      @extend .table-head-cell;

      &.column-name {
         font-weight: bold;
      }
   }
}
