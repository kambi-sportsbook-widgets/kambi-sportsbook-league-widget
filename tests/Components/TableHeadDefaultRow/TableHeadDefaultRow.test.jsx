/* eslint-env jest */
import React, { Children } from 'react';
import TableHeadDefaultRow from '../../../src/js/Components/TableHeadDefaultRow/TableHeadDefaultRow';
import ReactTestRenderer from 'react-test-renderer';

jest.mock('kambi-widget-core-library', () => ({
   translationModule: {
      getTranslation: (key) => `Translated: "${key}"`
   }
}));

describe('TableHeadDefaultRow DOM rendering', () => {

   it('renders correctly when not collapsable', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDefaultRow columns={[{name: 'Test column 1', short: 'TC1'}]} collapsable={false} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly when collapsable', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDefaultRow columns={[{name: 'Test column 1', short: 'TC1'}]} collapsable={true} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with title', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDefaultRow
            columns={[{name: 'Test column 1', short: 'TC1'}]}
            collapsable={true}
            title="Test title"
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly in hidden mode', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDefaultRow
            columns={[{name: 'Test column 1', short: 'TC1'}]}
            collapsable={true}
            hiddenMode={true}
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly without columns', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDefaultRow
            columns={[]}
            collapsable={true}
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with many columns', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDefaultRow
            columns={[{name: 'Test column 1', short: 'TC1'}, {name: 'Test column 2', short: 'TC2'}]}
            collapsable={true}
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
