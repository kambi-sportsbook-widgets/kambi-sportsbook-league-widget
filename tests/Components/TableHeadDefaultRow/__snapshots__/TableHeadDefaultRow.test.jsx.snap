// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadDefaultRow DOM rendering renders correctly in hidden mode 1`] = `
<tr
  className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general"
>
  <th
    className="title"
    colSpan="2"
  />
  <th
    className="column-name"
    title="Translated: \\"TC1\\""
  />
  <th
    className="margin"
  />
</tr>
`;

exports[`TableHeadDefaultRow DOM rendering renders correctly when collapsable 1`] = `
<tr
  className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general"
>
  <th
    className="title"
    colSpan="2"
  />
  <th
    className="column-name"
    title="Translated: \\"TC1\\""
  >
    Translated: "TC1"
  </th>
  <th
    className="margin"
  />
</tr>
`;

exports[`TableHeadDefaultRow DOM rendering renders correctly when not collapsable 1`] = `
<tr
  className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general"
>
  <th
    className="title"
    colSpan="2"
  />
  <th
    className="column-name"
    title="Translated: \\"TC1\\""
  >
    Translated: "TC1"
  </th>
  <th
    className="margin"
  />
</tr>
`;

exports[`TableHeadDefaultRow DOM rendering renders correctly with many columns 1`] = `
<tr
  className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general"
>
  <th
    className="title"
    colSpan="2"
  />
  <th
    className="column-name"
    title="Translated: \\"TC1\\""
  >
    Translated: "TC1"
  </th>
  <th
    className="column-name"
    title="Translated: \\"TC2\\""
  >
    Translated: "TC2"
  </th>
  <th
    className="margin"
  />
</tr>
`;

exports[`TableHeadDefaultRow DOM rendering renders correctly with title 1`] = `
<tr
  className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general"
>
  <th
    className="title"
    colSpan="2"
  >
    Test title
  </th>
  <th
    className="column-name"
    title="Translated: \\"TC1\\""
  >
    Translated: "TC1"
  </th>
  <th
    className="margin"
  />
</tr>
`;

exports[`TableHeadDefaultRow DOM rendering renders correctly without columns 1`] = `
<tr
  className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general"
>
  <th
    className="title"
    colSpan="2"
  />
  <th
    className="margin"
  />
</tr>
`;
